### ISSUE TITLE
Training/finetuning D0 starting from tf_efficientdet0 weights leads to NaN values in the parameter gradients at a certain point.

### ISSUE THREAD

**Issue by baptist** [rwightman/efficientdet-pytorch#91]
When trying to finetune the efficientdet d0 model using (the tensorflow based) [pretrained weights for that specific model](https://github.com/rwightman/efficientdet-pytorch/releases/download/v0.1/tf_efficientdet_d0-d92fd44f.pth) the loss and model outputs go to NaN values at random moments.

I do not observe this behaviour in the case of doing exactly the same but using the D1 model (and according D1 tf based weights). I also do not observe this behaviour when starting from unpretrained models. This issue also seems to be independent of my learning rate size.

I was mostly wondering if this a known issue? am I missing something particular here?
Many thanks in advance!

**Comment by Ekta246** [comment]
> When trying to finetune the efficientdet d0 model using (the tensorflow based) [pretrained weights for that specific model](https://github.com/rwightman/efficientdet-pytorch/releases/download/v0.1/tf_efficientdet_d0-d92fd44f.pth) the loss and model outputs go to NaN values at random moments.
> 
> I do not observe this behaviour in the case of doing exactly the same but using the D1 model (and according D1 tf based weights). I also do not observe this behaviour when starting from unpretrained models. This issue also seems to be independent of my learning rate size.
> 
> I was mostly wondering if this a known issue? am I missing something particular here?
> Many thanks in advance!

Could you please tell me how do you exactly fine-tune. When I do the fine-tuning it gives, me a tensor mismatch error in class_net in spite of changing the # classes according to my custom dataset in the head.
